Sources/Danger/GitLabDSL.swift

```diff
@@ -0,0 +1,222 @@
+import Cocoa
+
+public struct GitLab: Decodable {
+    public enum CodingKeys: String, CodingKey {
+        case mergeRequest = "mr"
+        case metadata
+    }
+
+    public let mergeRequest: GitLabMergeRequest
+    public let metadata: GitLabMetadata
+}
+
+public struct GitLabMetadata: Decodable, Equatable {
+    public let pullRequestID: String
+    public let repoSlug: String
+}
+
+public struct GitLabMergeRequest: Decodable, Equatable {
+    public enum State: String, Decodable {
+        case closed
+        case locked
+        case merged
+        case open
+    }
+
+    public struct Milestone: Decodable, Equatable {
+        public enum CodingKeys: String, CodingKey {
+            case createdAt = "created_at"
+            case description
+            case dueDate = "due_date"
+            case id
+            case iid
+            case projectId = "project_id"
+            case startDate = "start_date"
+            case state
+            case title
+            case updatedAt = "updated_at"
+            case webUrl = "web_url"
+        }
+
+        public enum State: String, Decodable {
+            case closed
+        }
+
+        public let createdAt: Date
+        public let description: String
+        public let dueDate: Date
+        public let id: Int
+        public let iid: Int
+        public let projectId: Int
+        public let startDate: Date
+        public let state: State
+        public let title: String
+        public let updatedAt: Date
+        public let webUrl: String
+    }
+
+    public struct TimeStats: Decodable, Equatable {
+        public enum CodingKeys: String, CodingKey {
+            case humanTimeEstimate = "human_time_estimate"
+            case humanTimeSpent = "human_total_time_spent"
+            case timeEstimate = "time_estimate"
+            case totalTimeSpent = "total_time_spent"
+        }
+
+        public let humanTimeEstimate: Int?
+        public let humanTimeSpent: Int?
+        public let timeEstimate: Int
+        public let totalTimeSpent: Int
+    }
+
+    struct UserMergeData: Decodable, Equatable {
+        enum CodingKeys: String, CodingKey {
+            case canMerge = "can_merge"
+        }
+
+        let canMerge: Bool
+    }
+
+    public struct DiffRefs: Decodable, Equatable {
+        enum CodingKeys: String, CodingKey {
+            case baseSha = "base_sha"
+            case headSha = "head_sha"
+            case startSha = "start_sha"
+        }
+
+        let baseSha: String
+        let headSha: String
+        let startSha: String
+    }
+
+    public struct Pipeline: Decodable, Equatable {
+        public enum Status: String, Decodable {
+            case success
+        }
+
+        public enum CodingKeys: String, CodingKey {
+            case id
+            case ref
+            case sha
+            case status
+            case webUrl = "web_url"
+        }
+
+        public let id: Int
```

Comment:
<!--
  0 failure: 
  1 warning:  Variable name sho...
  
  
  DangerID: danger-id-Danger;
  File: Sources/Danger/GitLabDSL.swift;
  Line: 105;
-->

- :warning: Variable name should be between 3 and 40 characters long: 'id' (`identifier_name`)